src/openrct2/drawing/string.c

```diff
@@ -798,7 +799,44 @@ static void ttf_draw_string_raw_ttf(rct_drawpixelinfo *dpi, const utf8 *text, te
                         if (info->flags & TEXT_DRAW_FLAG_INSET) {
                             *(dst + width + dstScanSkip + 1) = info->palette[3];
                         }
-                        *dst = colour;
+
+                        if (*src > 180 || !gConfigFonts.enable_hinting)
+                        {
+                            // Centre of the glyph: use full colour.
+                            *dst = colour;
+                        }
+                        else if (*dst < 10 && *src > 50)
+                        {
+                            // For black backgrounds, hint using grey pixels.
+                            *dst = ColourMapA[COLOUR_WHITE].colour_0;
+                        }
+                        else if (*src > 70)
+                        {
+                            // Simulate font hinting by shading the background colour instead.
+                            if (info->flags & TEXT_DRAW_FLAG_OUTLINE)
+                            {
+                                // As outlines are black, these texts should always use a darker shade
+                                // of the foreground colour for font hinting.
+                                *dst = ColourMapA[COLOUR_WHITE].colour_1;
+                            }
+                            else if (colour == ColourMapA[COLOUR_DARK_GREEN].colour_11)
+                            {
+                                // Exception: theme window text colour.
+                                *dst = ColourMapA[COLOUR_DARK_GREEN].lightest;
+                            }
+                            else if (colour == ColourMapA[COLOUR_BRIGHT_RED].light)
+                            {
+                                // Exception: losses in finance window.
+                                *dst = ColourMapA[COLOUR_BRIGHT_RED].lightest;
+                            }
+                            else
+                            {
+                                // For other texts, hinting direction depends on the intensity of the
+                                // foreground colour.
+                                bool intensive = colour % 12 > 4 && colour % 12 < 10;
```

Comment:
Do these numbers mean anything, or are they the result of experimentation?

Reply:
`colour % 12` gets you the palette row in question, as you probably figured. I found the domain between 4 and 10 through experimenting — these seem to work reasonably well across the board, save for the few exceptions as implemented.